### PR TITLE
Add isort and black support.

### PR DESCRIPTION
**Pull request by Natim**
Based on #443 

Add isort and black support.

### CODE REVIEW

**Comment by brunobord** [comment]
TL;DR: We do not intend to use black or isort on this project.

### A - `isort`

This library sorts imports **by alphabetical order**, and numerous imports in several files are made in "logical order". For example, all the states of a country before the other countries of a continent. I think that sorting in alphabetical order doesn't give any interesting value and may eventually be confusing when it comes to understand which import(s) is/are missing.

For example, the following code:

```python
from workalendar.core import MON, TUE, WED, THU, FRI, SAT, SUN
```

would become:

```python
from workalendar.core import FRI, MON, SAT, SUN, THU, TUE, WED
```

Where's the logic here?

(And yes, I know you can surely skip this line as we can do for other linters, but what would be the point of having linters if it's to skip them, right?)

### B - `black`

1. black is by definition a **very opinionated** code formatter. PEP8 is a designated, long-standing *standard* that encompasses opinions and communities. Writing PEP8 compliant code is in the DNA of Pythonistas and even if it's one of the facets of it, it's one of the pillar of what's called "Pythonic code".
2. And from my experience, I doubt that it significantly improves readability, especially on function arguments, or long lists, that are indented and put on several lines instead of being, as above, trimmed and grouped in "logical order". Of course, this argument is subjective, but stating that using black improves readability is also subjective.
3. On this project, it has occurred several times that contributors are having hard times fulfilling the simple PEP8 compliance. And I'm convinced that one can write PEP8-compliant code without having to use an external tool to do so. For black, as it's not (yet) a standard or a cultural habit, I doubt you can produce black-ready files in one go.
4. Adding a commit that changes (almost?) 100% of the workalendar project files is not a good thing from a git history point of view. It would raise a "event horizon" wall in the logs. Let's say that I've found a bug in the code and I'm searching for the commit that induced this regression ; and that this commit has been made **before** this event horizon in the git timeline. I would have hard time figuring this out. I *know* that it's somehow possible through github interface, to some extent.

What I see with black is a "technological solutionism" issue: adding a _cool feature_ on one hand has a cost on the other, so the benefit is, in my opinion, not as positive as one could expect and it adds too much of an uncertainty on a codebase that lives for more than 6 years now.

I think it might be interesting to use black from scratch, on a brand new project, but for workalendar I think it'd be too much of a burden.

**Comment by Natim** [comment]
Make sense thank you.
